src/analysis/polynomial.lean

```diff
@@ -0,0 +1,25 @@
+import data.real.basic
+import data.polynomial.default
+import data.set.intervals.infinite
+
+section real
+variables (f : polynomial ℝ)
+open set polynomial
+
+lemma f_zero_on_interval_f_zero {a b : ℝ} (h : a < b) (f_zero: ∀ x ∈ Icc a b, f.eval x = 0) : f = 0 :=
```

Comment:
Please generalize this to polynomials over integral domains, and arbitrary infinite sets.